kerastuner/protos/kerastuner.proto

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2019 The KerasTuner Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+syntax = "proto3";
+
+package kerastuner;
```

Comment:
Since the main use case for this would be for internal integration, we should consider the proper format for these protos.

We already have a representation of HParams for Tensorboard at:
https://github.com/tensorflow/tensorboard/blob/2b5547fe5fb96eba08f14ed2a97b2f8d519423e3/tensorboard/plugins/hparams/api.proto
and example parsing code in:
https://github.com/tensorflow/tensorboard/blob/2b5547fe5fb96eba08f14ed2a97b2f8d519423e3/tensorboard/plugins/hparams/summary_v2.py

We should consider how specialized this proto should be for KerasTuner VS if this should be a more generic proto that can be used in Tensorboard and other config files; and the KerasTuner HyperParameters object can initialize from it. I'll follow up offline.